src/key.rs

```diff
@@ -294,11 +294,17 @@ pub async fn store_self_keypair(
                 KeyPairUse::ReadOnly => false,
             };
 
+            // `addr` and `is_default` written for compatibility with older versions,
+            // until new cores are rolled out everywhere.
+            // otherwise "add second device" or "backup" may break.
+            // moreover, this allows downgrades to the previous version.
+            // writing of `addr` and `is_default` can be removed ~ 2024-08
+            let addr = keypair.addr.to_string();
```

Comment:
Is `to_string()` is really needed here?

Reply:
idk -  i did not wanted to optimise code but just do what was done before to protect against further side effects :)